### ISSUE TITLE
No 'Access-Control-Allow-Origin' header is present on the requested resource

### ISSUE THREAD

**Issue by taobad** [lejard-h/google_maps_webservice#70]
Hello, i am getting this error when i send a request to maps/api/place.
I am developing for web. Thanks

Error: Access to fetch at 'https://maps.googleapis.com/maps/api/place/details/json?placeid=ChIJ70taCKKAhYAR5IMmYwQT4Ts&key={APIKEY}' from origin  has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

// dependencies
dependencies:
  flutter:
    sdk: flutter
  provider: ^3.1.0
  cupertino_icons: ^0.1.3
  cloud_firestore: ^0.13.4+2
  google_maps_flutter: ^0.5.25+3
  google_maps_webservice: ^0.0.16
  location: ^2.3.5

**Comment by clus90** [comment]
Same issue here! .. Please help, I even uploaded my app to Firebase Hosting, added the header correctly (I think) on the firebase.json. But it doesn't work.

**Comment by nmacherey** [comment]
Same issue here. Working on iOS and Android not in web

**Comment by gustav3d** [comment]
Use a proxy or the js lib.

**Comment by Kuhno92** [comment]
is there an example code on how to use the js lib inside flutter? can't get it to work right now :(

**Comment by gustav3d** [comment]
https://pub.dev/packages/google_maps


> is there an example code on how to use the js lib inside flutter? can't get it to work right now :(

https://pub.dev/packages/google_maps

**Comment by Dome81** [comment]
Hello, I put <script src="https://maps.googleapis.com/maps/api/js?key=<my_key>&libraries=places"></script> inside index.html but I still have cors error. No problems with the android emulator. What am I doing wrong?

**Comment by andrea689** [comment]
I have found a workaround!

```
GoogleMapsPlaces(
          apiKey: 'YOUR_API_KEY',
          baseUrl: kIsWeb
              ? 'https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api'
              : null,
        ),
```

**Comment by EricPHassey** [comment]
Does anyone else have a better solution to this. cors-anywhere.herokuapp.com has placed some strict methods to stop the "abuse" of misintentional using their site. What is the preferred solution to resolve this.

**Comment by jayant0jhaSanskrut** [comment]
@EricPHassey though I would not personally recommend this as I feel like it's a temporary fix  but you can deploy your own cors anywhere server. [https://stackoverflow.com/a/47085173](Check this)

**Comment by jayant0jhaSanskrut** [comment]
Still facing this issue.

**Comment by lejard-h** [comment]
It will probably never work on a browser because of Google API limitation. Having a proxy server is the only solution.

**Comment by benji-farquhar** [comment]
> I have found a workaround!
> 
> ```dart
> GoogleMapsPlaces(
>   apiKey: 'YOUR_API_KEY',
>   baseUrl: kIsWeb
>       ? 'https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api'
>       : null,
> );
> ```

I am new to proxy servers. Does the Heroku solution cost money? My Flutter app already has a custom ASP.NET Core back end. Does [this](https://stackoverflow.com/a/54149906/3935156) essentially do the same thing, and if it is working in the back end, could it be used from my Flutter app with the same workaround by @andrea689?

**Comment by benji-farquhar** [comment]
> use a proxy or the js lib.

@gustav3d Do you mean the [google maps javascript library](https://developers.google.com/maps/documentation/javascript/libraries)?

**Comment by benji-farquhar** [comment]
I got mine working with an asp.net reverse proxy library called Yarp. [Here is the code](https://stackoverflow.com/a/76697909/3935156). 

My Flutter code is similar to the above solution; however the baseUrl is just my server address (localhost for now), so it may differ very slightly depending on your reverse proxy's way of doing things:

```
GoogleMapsPlaces(
        apiKey: EnvironmentConstants.of().placesApiKey,
        baseUrl: kIsWeb ? 'http://xxx.xxx.x.xx:5002' : null,
      ),
```

So yeah, you can create a reverse proxy in any back end if you already have one (difficulty will vary), and the Flutter portion of the solution will be _roughly_ the same. It might just take some learning on the reverse proxy side of things. 

This solution is the most financially scalable for large apps, as you have unlimited requests for free on localhost, and when you go live, you won't be starting up a whole new plan with a hosting company like Heroku (their prices and request limits are liable to change over time, I'm pretty sure it is not unlimited for free, maybe it used to be), it's just part of your back-end app.

**Comment by micky113** [comment]
if using flutter for web, first download FlutterGooglePlacesWeb, get GooglePlacesApi Key,then write the below code and test 
Also, allow temporary access to CORS anywhere Proxy by visiting [https://cors-anywhere.herokuapp.com/corsdemo]

 FlutterGooglePlacesWeb(
                apiKey: kGoogleApiKey,
                proxyURL: 'https://cors-anywhere.herokuapp.com/',
                required: true,
              ),
              TextButton(
                onPressed: () {
                  print(FlutterGooglePlacesWeb.value['name']); // '1600 Amphitheatre Parkway, Mountain View, CA, USA'
                  print(FlutterGooglePlacesWeb.value['streetAddress']); // '1600 Amphitheatre Parkway'
                  print(FlutterGooglePlacesWeb.value['city']); // 'CA'
                  print(FlutterGooglePlacesWeb.value['country']);
                  setState(() {
                    test = FlutterGooglePlacesWeb.value['name'] ?? '';
                  });
                },
                child: Text('Press to test'),
              ),
              Text(test),
